### PR TITLE
Finished fixing support for OS X

### PR DESCRIPTION
**Pull request by LRFLEW**
This version has been tested to work on 10.11 without any noticeable problems. 

This includes some commits from #2401 that were necessary to get it working (but not the main travis commits that the PR was mainly about), so it will likely need a rebase before merging.

### CODE REVIEW

File: src/openrct2.c [hunk 1]
```diff
@@ -610,30 +610,6 @@ static bool openrct2_release_rct2_segment()
 {
 	bool result = true;
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
-	int len = 0x01429000 - 0x8a4000; // 0xB85000, 12079104 bytes or around 11.5MB
```
Comment (duncanspumpkin) [inline src/openrct2.c]: Why is this being removed for linux? It still uses the mmap technique to setup the segments.

Comment (LRFLEW) [inline src/openrct2.c]: I'm not sure. I'll look through the commit history to see where that happened.

**Comment by janisozaur** [comment]
A quick test showed it launches on Linux, but I'd like to take a closer look in the evening (T-11h).

One comment about the PR itself: in one of the commits you introduce `openrct2_data` and `openrct2_text`. While I understand reason for this and net change is the same, this file will still live forever in git history. I think we would be better off without it ever reaching the tree in the first place.

**Comment by janisozaur** [comment]
And one more comment regarding extracting the segments: it should be made into a separate target, upon which `openrct2` should depend.

**Comment by janisozaur** [comment]
Rebase on top of #2457, the bits @LinusU extracted from #2401 needed for this to work.

**Comment by janisozaur** [comment]
@LRFLEW also use updated #2447, you seem to have resolved it wrong here and ended up redeclaring `err`

File: src/openrct2.c [hunk 2]
```diff
@@ -634,7 +634,7 @@ static bool openrct2_release_rct2_segment()
 		log_error("Failed to close file! errno = %d", err);
 		result = false;
 	}
-#endif // defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+#endif
```
Comment (janisozaur) [inline src/openrct2.c]: Can you leave a comment here?

Comment (LRFLEW) [inline src/openrct2.c]: Whoops, my bad. I'll fix that.

**Comment by janisozaur** [comment]
As far as I'm concerned it looks good and I tested it on Linux. :+1:

**Comment by LinusU** [comment]
Just tried latest `develop` out and it works perfectly!! Thanks a ton to all involved :fireworks: :balloon: :beers:

**Pull request merged**